src/battle_util.c

```diff
@@ -1256,6 +1256,11 @@ u8 DoFieldEndTurnEffects(void)
             gBattleStruct->turnCountersTracker++;
             break;
         case ENDTURN_GRASSY_TERRAIN:
+            if (gFieldStatuses & STATUS_FIELD_GRASSY_TERRAIN)
```

Comment:
I don't see it working in the last turn of Grassy Terrain. I think you should combine the two battlescripts into one.